### DISCUSSION TITLE
Rules compilation on windows, compiled rules used on iOS

### DISCUSSION THREAD

**Discussion by epouliquen** [microsoft/RulesEngine#134] [Q&A]
Hi,

we are very interested in your RulesEngine possibilities, but our use case is very special and I would like your point of view on what is possible or not.

We are developing a Unity3D app based on Visual editors running on Windows desktop, and on the other hand an iOS app, using data edited on windows, through a web-hosted database.

We have tried to use directly rules engines on the ios but of course we hit the 'compile' limitation for code generation on the type of aot platforms.

My question is : could we eventually think about split RulesEngine  behavior in two steps :

on windows use it to compile rules and get a factory
then find a way to serialize this factory instance (as dylib ?), put it on database
then on iOS deserialize it and continue using compiled rules to avoid code generation on this platform

Hope this strange use case is clear and hope you can help us :)

**Comment by abbasc52** [comment]
Currently it is not possible to generate serialized rules as deserializing it to code would be same as compilation. However internally we use Expression.Compile, which supports AOT platforms by switching to reflection-based interpretation instead of compiling.

We could add a config to enable interpretation by default. However, it would be difficult for us to test it as we don't support AOT as of now
